build_tools/bazel_to_cmake/bazel_to_cmake_converter.py

```diff
@@ -112,7 +112,10 @@ def _convert_target_block(self, name, target):
         target = target.replace("::", "_")
         return self._convert_string_arg_block(name, target, quote=False)
 
-    def _convert_srcs_block(self, srcs):
+    def _convert_srcs_block(self, srcs, name="SRCS", allowed_suffix=None):
+        if allowed_suffix:
+            srcs = [s for s in srcs if s.endswith(allowed_suffix)]
```

Comment:
Looks like "allowed_suffix" gives this a way for the filegroups here to not be converted?

https://github.com/openxla/iree/blob/6b93f11c7f19cb4810af533ec56ec07e58ab4aaf/runtime/src/iree/builtins/ukernel/arch/arm_64/BUILD.bazel#L28-L37

I'd like to avoid a patch like "allowed_suffix", or at least call out the partially implemented conversion explicitly somewhere. This sort of code can otherwise be difficult to maintain as the intent behind the code and original problem it solves are both obscured.

Maybe we should move away from using filegroups or find some more direct and possibly verbose way to model them across both build systems.

https://github.com/openxla/iree/blob/8d7dc80a2282987dc65c93452af5712cdfd40aa9/runtime/src/iree/builtins/ukernel/BUILD.bazel#L28-L46

---

Aside: we should try to keep the comments in those files up-to-date. If this PR handles things in CMake that were previously only handled in Bazel that should be clear.

Reply:
There is also some useful context in https://github.com/openxla/iree/blob/6bbdb7278aebf92a3f66f35b787ebb879886f5ea/build_tools/bazel_to_cmake/bazel_to_cmake_converter.py#L256-L263